### PR TITLE
feat: issue #61 added test to improve coverage of jaccard_similarity()

### PR DESCRIPTION
**Pull request by bastienfaivre**
Add tests and also documentation since I found some bug in the algorithm

### CODE REVIEW

File: complex_functions/tests/jaccard_similarity_test.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+# tests the algorithm on 2 sets or tuples with alternative union
+def test_two_lists_or_tuples_True():
+    setA = ('c', 'd', 'e', 'f', 'h', 'i')
+    setB = ('c', 'd', 'e', 'f', 'h', 'i')
+    # this test fails due to an uncovered case in the algorithm
+    # when the alternative union is chosen, union is a number
+    # and we try to do len(union) with is not valid
+    # assert jaccard_similariy(setA, setB, True) == 0
+    # assert jaccard_similariy(setB, setA, True) == 0
```
Comment (AmandaStromdahl) [inline complex_functions/tests/jaccard_similarity_test.py:35]: Should we run asserts to show that the algorithm has a bug? Wouldn't that also improve the coverage?

File: complex_functions/tests/jaccard_similarity_test.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    assert jaccard_similariy(0, (0)) == None
+    assert jaccard_similariy(0, 0, True) == None
+    assert jaccard_similariy((0), 0, True) == None
+    assert jaccard_similariy(0, (0), True) == None
+    # assert jaccard_similariy((), (), True) == ?
+    # assert jaccard_similariy((), (0), True) == ?
+    # assert jaccard_similariy((0), (), True) == ?
+    # assert jaccard_similariy([], []) == ?
+    # assert jaccard_similariy([], [0]) == ?
+    # assert jaccard_similariy([0], []) == ?
```
Comment (AmandaStromdahl) [inline complex_functions/tests/jaccard_similarity_test.py]: Can we remove these comments?

File: complex_functions/tests/jaccard_similarity_test.py (hunk 1 above)
Comment (bastienfaivre) [reply complex_functions/tests/jaccard_similarity_test.py:35]: The test will fail so I don't know if it would be a valid coverage

**Comment by AmandaStromdahl** [comment]
Overall I think it looks good, just have some questions above.

File: complex_functions/tests/jaccard_similarity_test.py (hunk 2 above)
Comment (bastienfaivre) [reply complex_functions/tests/jaccard_similarity_test.py]: As I mentioned above the test, the behavior of these cases aren't described in the doc, but if we follow the comprehension of the algorithm, this should be invalid inputs, but the algorithm seems to accept it. So what do you think ? Should I removed it or let it like I did ?

Comment (AmandaStromdahl) [reply complex_functions/tests/jaccard_similarity_test.py]: I think it would look nicer to remove those comments.

Comment (bastienfaivre) [reply complex_functions/tests/jaccard_similarity_test.py]: ok I'll do it

File: complex_functions/tests/jaccard_similarity_test.py (hunk 1 above)
Comment (AmandaStromdahl) [reply complex_functions/tests/jaccard_similarity_test.py:35]: I see, in that case we can keep it as is.

**Comment by AmandaStromdahl** [comment]
Looks good.

**Pull request merged**